### PR TITLE
Front sim branch

### PR DESCRIPTION
**Pull request by adithyasampath**
Updated branch with new dataloader for by using ConcatDataset. Updated the config file for n1 (train_front.yaml). Removed all unnecessary scripts. Added 2 test scripts - 1) one to check what are the unique labels present in the seg dataset folder (to check if they match nuscenes/carla labels) and 2) one to check if concatdataset is working fine.

### CODE REVIEW

**Review by punit-bhatt — CHANGES_REQUESTED** [review]

File: nuscenes_labels.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
```
[the remark this answers is not in the document]
Comment (adithyasampath) [reply nuscenes_labels.json]: Removed file

File: test_dataloader.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import os
```
[the remark this answers is not in the document]
Comment (adithyasampath) [reply test_dataloader.py:1]: Added in my gitignore. Shouldn't be there now

File: visualize.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
+import numpy as np
```
[the remark this answers is not in the document]
Comment (adithyasampath) [reply visualize.py]: Moved to utils

File: dataloaders/collect_nuscenes_dataset.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+import zipfile
+import cv2
+import shutil
+import argparse
+from tqdm import tqdm
+from carla_nuscenes_map import NUSCENES_CARLA_MAP
+
+
+def create_dir(folder):
+      if not os.path.isdir(folder):
```
[the remark this answers is not in the document]
Comment (adithyasampath) [reply dataloaders/collect_nuscenes_dataset.py:15]: Corrected.

File: dataloaders/test_segmentation_labels.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
+import cv2
```
[the remark this answers is not in the document]
Comment (adithyasampath) [reply dataloaders/test_segmentation_labels.py:1]: Moved to utils

File: dataloaders/carla_nuscenes_map.py [hunk 6]
```diff
@@ -51,6 +51,7 @@
     22: (145, 170, 100)
 }
 
+
```
[the remark this answers is not in the document]
Comment (adithyasampath) [reply dataloaders/carla_nuscenes_map.py:54]: Didn't get you?

File: train_n1.py [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+from losses import get_loss_fn
+from metrics import get_metrics
+from models import get_network
+from collections import OrderedDict
+from omegaconf import DictConfig
+from train_utils import train_and_evaluate
+
+import wandb
+
+wandb.init(project="vlr_project", entity="adithyasampath21")
```
[the remark this answers is not in the document]
Comment (adithyasampath) [reply train_n1.py]: Removed wandb

File: visualize.py [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+                                  ".",
+                                  ckpt_filename)
+    model.eval()
+    
+    # nuscenes mini
+    # file_list = ["000050", "000052", "000053", "000088"]
+    
+    # carla mini
+    file_list = ["002110", "002180", "002190", "002200", "002210", "002220", "002230", "002230", "002240", "002250", "002260", "002270"]
+
```
[the remark this answers is not in the document]
Comment (adithyasampath) [reply visualize.py]: Removed. They take args from argeparse now

File: visualize.py [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+    
+    # nuscenes mini
+    # file_list = ["000050", "000052", "000053", "000088"]
+    
+    # carla mini
+    file_list = ["002110", "002180", "002190", "002200", "002210", "002220", "002230", "002230", "002240", "002250", "002260", "002270"]
+    
+    img_format = "/home/adithyas/atdt/results/{}_rgb.png"
+    save_format = "/home/adithyas/atdt/results/{}_result.png"
+    img_paths = [img_format.format(file) for file in file_list]
```
[the remark this answers is not in the document]
Comment (adithyasampath) [reply visualize.py]: Removed

File: train_n2.py [hunk 10]
```diff
@@ -25,7 +25,7 @@ def main(cfg: DictConfig):
     params = cfg.training
     ckpt_filename = "checkpoint.pt"
     log_dir = os.path.join(cfg.logging.log_dir, f'{cfg.model_name}/{cfg.exp}/')
-    ckpt_dir = os.path.join(cfg.logging.ckpt_dir, f'{cfg.model_name}/{cfg.exp}')
+    ckpt_dir = os.path.join(cfg.logging.log_dir, f'{cfg.model_name}/{cfg.exp}')
```
[the remark this answers is not in the document]
Comment (adithyasampath) [reply train_n2.py]: Corrected
